File: Share/src/main/java/com/buguagaoshu/share/domain/FileMessage.java
package com.buguagaoshu.share.domain;

import lombok.Data;

import jakarta.persistence.*;

/**
 * @author Pu Zhiwei {@literal puzhiweipuzhiwei@foxmail.com}
 * create          2020-07-03 15:22
 */
@Data
@Entity
@Table(name = "file_table")
public class FileMessage {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    /**
     * 原文件名
     * */
    private String uploadFilename;

    /**
     * 存储的文件名
     * */
    private String filename;

    private Long size;

    /**
     * 保存路径
     * */
    private String path;

    /**
     * 文件所属路径时间
     * */
    private String date;


    private Long createTime;
}

File: Go/internal/model/file_message.go
package model

// FileMessage 对齐 Spring com.buguagaoshu.share.domain.FileMessage（表 file_table）
type FileMessage struct {
	ID                     *int64 `gorm:"primaryKey;autoIncrement;column:id" json:"id"`
	UploadFilename         string `gorm:"column:upload_filename" json:"uploadFilename"`
	Filename               string `gorm:"column:filename" json:"filename"`
	Size                   *int64 `gorm:"column:size" json:"size"`
	Path                   string `gorm:"column:path" json:"path"`
	Date                   string `gorm:"column:date" json:"date"`
	CreateTime             *int64 `gorm:"column:create_time" json:"createTime"`
	PublicUser             *bool  `gorm:"column:public_user" json:"publicUser"`
	Url                    string `gorm:"column:url" json:"url"`
	UserSeeKey             string `gorm:"column:user_see_key" json:"userSeeKey"`
	HaveUserSeeKey         bool   `gorm:"column:have_user_see_key;default:false" json:"haveUserSeeKey"`
	CreatePublicShareUserId *int64 `gorm:"column:create_public_share_user_id" json:"createPublicShareUserId"`
}

func (FileMessage) TableName() string { return "file_table" }
